libraries/AP_Compass/AP_Compass.h

```diff
@@ -124,6 +127,7 @@ friend class AP_Compass_Backend;
     // compass calibrator interface
     void cal_update();
 
+#if COMPASS_MOT_ENABLED
```

Comment:
This triggers a crazy number of warnings when compiling:
```
In file included from ../../libraries/AP_AHRS/AP_AHRS.h:25,
                 from ../../libraries/AP_Avoidance/AP_Avoidance.cpp:6:
../../libraries/AP_Compass/AP_Compass.h:130:5: warning: this use of "defined" may not be portable [-Wexpansion-to-defined]
  130 | #if COMPASS_MOT_ENABLED
      |     ^~~~~~~~~~~~~~~~~~~
../../libraries/AP_Compass/AP_Compass.h:257:5: warning: this use of "defined" may not be portable [-Wexpansion-to-defined]
  257 | #if COMPASS_MOT_ENABLED
      |     ^~~~~~~~~~~~~~~~~~~
In file included from ../../libraries/AP_AHRS/AP_AHRS.h:25,
                 from ../../libraries/AP_Avoidance/AP_Avoidance.cpp:6:
../../libraries/AP_Compass/AP_Compass.h:356:5: warning: this use of "defined" may not be portable [-Wexpansion-to-defined]
  356 | #if COMPASS_CAL_ENABLED
      |     ^~~~~~~~~~~~~~~~~~~
../../libraries/AP_Compass/AP_Compass.h:477:5: warning: this use of "defined" may not be portable [-Wexpansion-to-defined]
  477 | #if COMPASS_CAL_ENABLED
      |     ^~~~~~~~~~~~~~~~~~~
../../libraries/AP_Compass/AP_Compass.h:481:5: warning: this use of "defined" may not be portable [-Wexpansion-to-defined]
  481 | #if COMPASS_MOT_ENABLED
      |     ^~~~~~~~~~~~~~~~~~~
```

Reply:
which compiler and target is this with? I don't get these warnings

Reply:
STIL, gcc (GCC) 9.1.0